src/components/IdeasContainer.jsx

```diff
@@ -0,0 +1,81 @@
+'use client';
+import React, { useEffect, useState } from 'react';
+import { Button, Grid } from '@mui/material';
+import IdeaCardComponent from './IdeaCardComponent';
+
+const IdeasContainer = ({ ideas }) => {
+    const cardsPerPage = 5;
+    const [currentPage, setCurrentPage] = useState(0);
+    const [resolvedIdeas, setResolvedIdeas] = useState([]);
+    const [visibleIdeasSlice, setVisibleIdeasSlice] = useState([]);
+
+    useEffect(() => {
+        if (ideas instanceof Promise) {
```

Comment:
Why `ideas` is a promise? Can you resolve that promise before passing it to `useEffect`?

Reply:
I think it is done...